robusta_krr/core/integrations/prometheus/loader.py

```diff
@@ -92,65 +38,31 @@ def __init__(
-
-        self.url = self.config.prometheus_url
-        self.url = self.url or self.prometheus_discovery.find_prometheus_url()
-
-        if not self.url:
-            raise PrometheusNotFound(
-                f"Prometheus instance could not be found while scanning in {cluster or 'default'} cluster.\n"
-                "\tTry using port-forwarding and/or setting the url manually (using the -p flag.)."
-            )
-
-        self.info(f"Using prometheus at {self.url} for cluster {cluster or 'default'}")
-
-        headers = {}
-
-        if self.auth_header:
-            headers = {"Authorization": self.auth_header}
-        elif not self.config.inside_cluster:
-            self.api_client.update_params_for_auth(headers, {}, ["BearerToken"])
-
-        self.prometheus = CustomPrometheusConnect(url=self.url, disable_ssl=not self.ssl_enabled, headers=headers)
-        self._check_prometheus_connection()
-
-        self.info(f"Prometheus connected successfully for {cluster or 'default'} cluster")
-
-    def _check_prometheus_connection(self):
-        """
-        Checks the connection to Prometheus.
-
-        Raises:
-            PrometheusNotFound: If the connection to Prometheus cannot be established.
-        """
-
-        try:
-            response = self.prometheus._session.get(
-                f"{self.prometheus.url}/api/v1/query",
-                verify=self.prometheus.ssl_verification,
-                headers=self.prometheus.headers,
-                # This query should return empty results, but is correct
-                params={"query": "example"},
-            )
-            response.raise_for_status()
-        except (ConnectionError, HTTPError) as e:
-            raise PrometheusNotFound(
-                f"Couldn't connect to Prometheus found under {self.prometheus.url}\nCaused by {e.__class__.__name__}: {e})"
-            ) from e
-
-    async def query(self, query: str) -> dict:
-        return await asyncio.to_thread(self.prometheus.custom_query, query=query)
+        self.loader = self.get_metrics_service(config, api_client=self.api_client, cluster=cluster)
+        if not self.loader:
+            raise PrometheusNotFound("No Prometheus or metrics service found")
+
+        self.info(f"{self.loader.name()} connected successfully for {cluster or 'default'} cluster")
+
+
+    def get_metrics_service(self,
+            config: Config,
+            api_client: Optional[ApiClient] = None,
+            cluster: Optional[str] = None,) -> Optional[MetricsService]:
+        
+        for service_name, metric_service_class in METRICS_SERVICES.items():
+            try:
+                loader = metric_service_class(config, api_client=api_client, cluster=cluster)
+                loader.check_connection()
+                self.echo(f"{service_name} found")
+                return loader
+            except Exception as e:
+                self.warning(f"{service_name} not found")
```

Comment:
I'm not sure this should be a warning, rather the debug. Like: why should the used be warned that we can not find victoriametrics, if they are using thanos?